src/components/base/outline-modal/outline-modal.ts

```diff
@@ -0,0 +1,263 @@
+import { html, TemplateResult, CSSResultGroup } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import componentStyles from './outline-modal.css.lit';
+import { OutlineElement } from '../outline-element/outline-element';
+import { ifDefined } from 'lit/directives/if-defined';
+
+export type ModalSize = 'small' | 'medium' | 'full-screen';
+
+export const modalSizes: ModalSize[] = ['small', 'medium', 'full-screen'];
+
+// See https://stackoverflow.com/questions/1599660/which-html-elements-can-receive-focus.
+// @todo make this re-usable across components?
+const focusableElementSelector = `
```

Comment:
I don't know if you need this. This is describing default behaviors, and in the case of the modal I think you can provide prop that takes an element id that gets focus when the modal opens, and that defaults to the "#main" div wrapping your default slot.